### DISCUSSION TITLE
Error reading JPEG 2000 image

### DISCUSSION THREAD

**Discussion by MichaelTran262** [libvips/libvips#2387] [Q&A] [answered]
When I try to copy jp2 file to png or any other format in debug I get an error.
```
$ vips copy town.jp2 x.png
vips_foreign_load_jp2k_build:
vips_foreign_load_jp2k_header:
image:
x0 = 0, y0 = 0, x1 = 25540, y1 = 18799, numcomps = 3, color_space = 0
icc_profile_buf = (nil), icc_profile_len = 0
components:
0) dx = 1, dy = 1, w = 25540, h = 18799, x0 = 0, y0 = 0
    prec = 8, bpp = 0, sgnd = 0, resno_decoded = 0, factor = 0
    data = (nil), alpha = 0
1) dx = 1, dy = 1, w = 25540, h = 18799, x0 = 0, y0 = 0
    prec = 8, bpp = 0, sgnd = 0, resno_decoded = 0, factor = 0
    data = (nil), alpha = 0
2) dx = 1, dy = 1, w = 25540, h = 18799, x0 = 0, y0 = 0
    prec = 8, bpp = 0, sgnd = 0, resno_decoded = 0, factor = 0
    data = (nil), alpha = 0
info:
tx0 = 0, ty0 = 0, tdx = 4096, tdy = 4096, tw = 7, th = 5
nbcomps = 3, tile_info = (nil)
vips_foreign_load_jp2k_load:

(vips:245): VIPS-WARNING **: jp2kload: JP2 box which are after the codestream will not be read by this function.


(vips:245): VIPS-WARNING **: jp2kload: JP2 box which are after the codestream will not be read by this function.


(vips:245): VIPS-WARNING **: error in tile 4096 x 0
jp2kload: Invalid tile part index for tile number 1. Got 5, expected 0

jp2kload: Fail to read the current marker segment (0xff90)

jp2kload: Failed to decode the codestream in the JP2 file

vips_region_generate: stop requested
vips2png: unable to write to target x.png
vips_foreign_load_jp2k_dispose:
```
I understand that jp2kload can't read something. But when I `opj_decompress` command, it converts the file with no error. Where could be the problem?

**Comment by jcupitt** [comment, marked as answer]
Hi again, could you upload a sample file that fails?

**Reply by MichaelTran262** [reply]
Hello again too. Of course. Here is a zip archive with the sample image. The error is also in other images similar to the one in the archive.

[Google link](https://drive.google.com/file/d/1OknRTdEL-LoSOnDk4ufuWXPXIQYordeq/view?usp=sharing)

**Reply by jcupitt** [reply]
Thanks, I'll have a look when I'm back at work next week.

**Reply by MichaelTran262** [reply]
Thank you very much!

**Reply by MichaelTran262** [reply]
The error is also in these smaller jp2 files. Imagemagick can convert them, but libvips is returning the same error.

[Google link](https://drive.google.com/file/d/1yTT_PbtOeaecfWfpSoFdDqMtVtbGJDOL/view?usp=sharing)

**Reply by jcupitt** [reply]
Hi again, sorry, this dropped off my radar :( I agree this is a libnvips bug and I've made an issue:

https://github.com/libvips/libvips/issues/2621

Thanks for the report!

**Reply by jcupitt** [reply]
I investigated, and I think these images are out of spec:

https://github.com/libvips/libvips/issues/2621#issuecomment-1014637267

Sorry, I don't think there's anything libvips can do here.

**Reply by MichaelTran262** [reply]
That is alright. Thank you!
